### PR TITLE
proposal for cloud_hosts to use vault token from openstack vault plugin

### PR DESCRIPTION
**Pull request by kucerakk**

### CODE REVIEW

File: playbooks/roles/cloud_host/tasks/main.yaml [hunk 1]
```diff
@@ -1,4 +1,35 @@
...
+  register: "cloud_token"
+  when: "clouds[cloud].vault_path is defined"
+
+- name: "Combine cloud_auth with vault artifacts"
+  no_log: true
+  set_fact:
+    cloud_auth: "{{ cloud_auth | combine(vault_items, recursive=true) }}"
+  vars:
+    vault_items:
+      "auth":
```
Comment (gtema) [inline playbooks/roles/cloud_host/tasks/main.yaml:26]: actually target was always to use response of plugin as a ready clouds entry. So even here it should be sufficient to have vars: vault_items: auth: "{{ cloud_token.data.data }}"

File: playbooks/roles/cloud_host/tasks/main.yaml [hunk 2]
```diff
@@ -1,4 +1,22 @@
...
+    url: "{{ ansible_hashi_vault_addr }}"
+    token: "{{ ansible_hashi_vault_token }}"
+    path: "{{ clouds[cloud].vault_path }}"
+  register: "cloud_token"
+  when: "clouds[cloud].vault_path is defined"
+
+- name: "Set cloud_auth variable with auth artifacts from vault"
+  no_log: true
+  set_fact:
+    cloud_auth: "{{ cloud_token.data.data }}"
```
Comment (gtema) [inline playbooks/roles/cloud_host/tasks/main.yaml:18]: set_fact is bad cause it is stored in the facts cache. If this is really required (and here it is not really required) we can drop it and do cloud_token.data.data every time we need it. Alternatively we may want to unset fact at the end. This is not critical for us here, but generally we should really keep this in mind

File: playbooks/roles/cloud_host/tasks/main.yaml (hunk 1 above)
Comment (kucerakk) [reply playbooks/roles/cloud_host/tasks/main.yaml]: fixed

File: playbooks/roles/cloud_host/tasks/main.yaml (hunk 2 above)
Comment (kucerakk) [reply playbooks/roles/cloud_host/tasks/main.yaml]: fixed

**Review by gtema — APPROVED** [review]

**Pull request merged**